SU2_PY/SU2_Nastran/pysu2_nastran.py

```diff
@@ -943,24 +984,27 @@ def applyload(self, iVertex, fx, fy, fz):
     """
     This method can be accessed from outside to set the nodal forces.
     """
-
-    makerID = list(self.markers.keys())
-    makerID = makerID[0]
-    iPoint = self.getInterfaceNodeGlobalIndex(makerID, iVertex)
+    iPoint = self.getVertexGlobalIndex(self.FSI_marker, iVertex)
     self.node[iPoint].SetForce((fx,fy,fz))
 
+  def getNumberOfModes(self):
+    """
+    This method provides the number of degrees of freedom used in
+    the structural solver.
+    """
+    return self.nDof
+
   def getFSIMarkerID(self):
     """
     This method provides the ID of the interface marker
     """
-    L = list(self.markers)
-    return L[0]
+    return self.FSI_marker
 
   def getNumberOfSolidInterfaceNodes(self, markerID):
 
     return len(self.markers[markerID])
 
-  def getInterfaceNodeGlobalIndex(self, markerID, iVertex):
+  def getVertexGlobalIndex(self, markerID, iVertex):   # TODO This solver is serial, thus global=local
```

Comment:
Are the TODO on your list to be tackled next? Is parallel use presented in those setups? I would rather error-check and document the state then to write a TODO in the code

Reply:
Hi Tobi! Thank you very much for your time. Yes, you are right... TODOs in the code are ugly. They are definitely being tackled right now, this is why they are there, but I can remove them and just keep a note.